cmake/modules/yaml.cmake

```diff
@@ -303,12 +306,32 @@ function(yaml_set)
     string(JSON subjson GET "${json_content}" ${ARG_YAML_KEY})
     string(JSON index LENGTH "${subjson}")
     list(LENGTH ARG_YAML_LIST length)
-    math(EXPR stop "${index} + ${length} - 1")
     if(NOT length EQUAL 0)
-      foreach(i RANGE ${index} ${stop})
-        list(POP_FRONT ARG_YAML_LIST value)
-        string(JSON json_content SET "${json_content}" ${ARG_YAML_KEY} ${i} "\"${value}\"")
-      endforeach()
+      list(GET ARG_YAML_LIST 0 entry_0)
+      if(entry_0 STREQUAL MAP)
+        math(EXPR length "${length} / 2")
+        math(EXPR stop "${index} + ${length} - 1")
+        foreach(i RANGE ${index} ${stop})
+          list(POP_FRONT ARG_YAML_LIST argument)
+          if(NOT argument STREQUAL MAP)
+            message(FATAL_ERROR "")
+            message(FATAL_ERROR "${function}(${argument} ) is not valid at this position.\n"
+                    "Syntax is 'LIST MAP \"key1: value1.1, ...\" MAP \"key1: value1.2, ...\""
+            )
+          endif()
+          list(POP_FRONT ARG_YAML_LIST map_value)
+          string(REGEX REPLACE "[ ]*(:|,)[ ]*" "\"\\1\"" qouted_map_value "\"${map_value}\"")
```

Comment:
This does not function correctly in windows enviroments as the path contains colons.
```
foo: name: sensor_client, source-dir: C:/Users/BillyBob/zephyrGoodX/zephyrNew/sensor_client, type: MAIN
bar: "name":"sensor_client","source-dir":"C":"/Users/BillyBob/zephyrGoodX/zephyrNew/sensor_client","type":"MAIN"
CMake Error at C:/Users/BillyBob/zephyrGoodX/zephyrNew/cmake/modules/yaml.cmake:326 (string):
  string sub-command JSON failed parsing json string: * Line 1, Column 41

    Missing ',' or '}' in object declaration
```

Reply:
@tejlmand I am seeing downstream CI errors on MS Windows as well after this PR was merged (https://github.com/CANnectivity/cannectivity/actions/runs/13002578162/job/36263865484).

Reply:
We've reverted the changes from this PR

Reply:
@WilliamGFish This has been fixed here, both in functionality but also with extra test cases and value validation so that we get better error feedback than a `JSON failed parsing json string'